src/unleash-action.ts

```diff
@@ -0,0 +1,115 @@
+import { UnleashClient } from 'unleash-proxy-client';
+import { Metrics } from './metrics';
+
+interface ICreateUnleashActionOptions {
+    url: string;
+    clientKey: string;
+    appName: string;
+    environment: string;
+    context: Record<string, string>;
+    features?: string[];
+    variants?: string[];
+    setResult: (name: string, value: any) => void;
+}
+
+interface IUnleashActionOptions extends ICreateUnleashActionOptions {
+    client: UnleashClient;
+    metrics: Metrics;
+}
+
+export const createUnleashAction = async (
+    options: ICreateUnleashActionOptions,
+): Promise<void> => {
+    const client = createClient(options);
+    const metrics = createMetrics(options);
+    const action = new UnleashAction({ ...options, client, metrics });
+    await action.run();
+    await action.end();
+};
+
+const createMetrics = (options: ICreateUnleashActionOptions): Metrics => {
+    return new Metrics({
+        headerName: 'Authorization',
+        appName: options.appName,
+        url: options.url,
+        clientKey: options.clientKey,
+    });
+};
+
+const createClient = (options: ICreateUnleashActionOptions): UnleashClient => {
+    return new UnleashClient({
+        appName: options.appName,
+        url: options.url,
+        clientKey: options.clientKey,
+        environment: options.environment,
+        refreshInterval: 0,
+        metricsInterval: 0,
+        disableMetrics: true,
```

Comment:
Built in metrics don't work?

Reply:
It's an async scheduled job over which we have no control and can't await - we can't access the internal metrics either to force sending metrics synchronously. Option B I guess is to expand on what's built into Proxy SDK JS - but that doesn't feel 100% right either